packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/quanticSearchBoxInput.js

```diff
@@ -197,69 +198,81 @@ export default class QuanticSearchBoxInput extends LightningElement {
   }
 
   handleValueChange() {
-    this.sendInputValueChangeEvent(this.input.value, false);
+    this.sendInputValueChangeEvent(this.input.value);
   }
 
   onSubmit(event) {
     event.stopPropagation();
-    this.sendInputValueChangeEvent(this.input.value, false);
     this.sendSubmitSearchEvent();
     this.input.blur();
   }
 
-  handleKeyDownOnClearButton(event) {
-    if (event.key === keys.ENTER) {
-      // Ignore the next enter key press in the searchbox input to prevent submitting a search when we press enter on the clear button.
-      this.ignoreNextEnterKeyPress = true;
-    }
-  }
-
-  /**
-   * Prevent default behavior of enter key, on textArea, to prevent skipping a line.
-   * @param {KeyboardEvent} event
-   */
-  onKeydown(event) {
-    if (event.key === keys.ENTER && !event.shiftKey) {
-      event.preventDefault();
-    }
-  }
-
   /**
    * @param {KeyboardEvent} event
    */
-  onKeyup(event) {
+  onKeyDown(event) {
     switch (event.key) {
+      case keys.ESC:
+        this.input.removeAttribute('aria-activedescendant');
+        this.input.blur();
+        break;
+
+      case keys.ARROWRIGHT:
+        // eslint-disable-next-line no-case-declarations
+        const selectedOption =
+          this.suggestionListElement?.getCurrentSelectedValue?.();
+        this.sendInputValueChangeEvent(selectedOption.value);
+        this.input.setAttribute('aria-activedescendant', selectedOption.id);
+        break;
+
       case keys.ENTER:
         this.handleEnter(event);
         break;
+
       case keys.ARROWUP:
-        this.suggestionListElement?.selectionUp();
+        event.preventDefault();
```

Comment:
I'm specifically adding preventDefault in ArrowUp and Arrowdown because the default behaviour on an input  receiving a keydown event with these two keys is the following:


https://github.com/coveo/ui-kit/assets/86681870/1bfc4206-0630-402b-9a01-81ffa73fc9a7

Notice how the cursor is moving from the start of the input to its end, that's what I am preventing.